src/dqe_idx.erl

```diff
@@ -227,6 +228,25 @@ metrics(Collection) ->
 
 %%--------------------------------------------------------------------
 %% @doc
+%% Returns a list of metric path suffixes of `Depth' that are prefixed
+%% by the given probe `Prefix', which can also be empty.
+%%
+%% For example:
+%% metrics(<<"collection">>, [], 1) -> [<<"base">>].
+%% metrics(<<"collection">>, [<<"base">>], 1) -> [<<"cpu">>].
+%% metrics(<<"collection">>, [], 2) -> [[<<"base">>,<<"cpu">>]].
+%% @end
+%%--------------------------------------------------------------------
+-spec metrics(Collection::collection(), Prefix::metric(),
+              Depth::pos_integer()) ->
+                    {ok, Metric::metric()} |
```

Comment:
I think the spec and examples are still off.

From what I understand the spec should be `[metric()]` and the example

```erlang
%% metrics(<<"collection">>, [], 1) -> [[<<"base">>]].
%% metrics(<<"collection">>, [<<"base">>], 1) -> [[<<"cpu">>]].
%% metrics(<<"collection">>, [], 2) -> [[<<"base">>,<<"cpu">>]].
```